dlrover/python/master/watcher/k8s_watcher.py

```diff
@@ -90,18 +92,34 @@ def _convert_pod_event_to_node_event(event):
         # We only care about pod related events
         return None
 
+    metadata: client.V1ObjectMeta = evt_obj.metadata
+    job_name = metadata.labels[ElasticJobLabel.JOB_KEY]
+
     # Skip events of dlrover mater Pod
-    pod_type = evt_obj.metadata.labels[ElasticJobLabel.REPLICA_TYPE_KEY]
+    pod_type = metadata.labels[ElasticJobLabel.REPLICA_TYPE_KEY]
     if pod_type == NodeType.DLROVER_MASTER:
         return None
 
-    metadata: client.V1ObjectMeta = evt_obj.metadata
     rank = int(metadata.labels[ElasticJobLabel.RANK_INDEX_KEY])
     pod_id = int(metadata.labels[ElasticJobLabel.REPLICA_INDEX_KEY])
     pod_name = metadata.name
     host_name = evt_obj.spec.node_name
     host_ip = evt_obj.status.host_ip
 
+    # Skip event of pod deleted if the deleted pod already exist for
+    # the deleted pod have already been successfully recovered
+    if evt_type == NodeEventType.DELETED:
+        pod_labels = _get_pod_unique_labels(job_name, pod_type, rank)
+        any_existed_running_pod = any(
+            k8s_util.is_target_labels_equal(pod_labels, pod.metadata.labels)
+            for pod in pod_list
```

Comment:
The `pod_list` from line 192 may be outdated. We need to get pods by unique labels of the pod and check wether there are running pods.

Reply:
done